cxmon/src/mon_cmd.cpp

```diff
@@ -303,6 +307,256 @@ void binary_dump(void)
 
 
 /*
+ * Add Break Point
+ */
+void break_point_add(void)
+{
+	uintptr address;
+
+	if (mon_token == T_END || !mon_expression(&address)) {
+		mon_error("Expect break point in hexadecimal.");
+		return;
+	}
+
+	if (mon_token != T_END) {
+		mon_error("Too many arguments");
+		return;
+	}
+
+	BREAK_POINT_SET::iterator it;
+	// Save break point
+	if ((it = disabled_break_points.find(address)) == disabled_break_points.end())
+		active_break_points.insert(address);
+	else {
+		disabled_break_points.erase(it);
+		active_break_points.insert(address);
+	}
+}
+
+
+bool validate_index(uintptr *index_ptr, const BREAK_POINT_SET& break_point_set)
+{
+	if (mon_token == T_END || !mon_expression(index_ptr)) {
+		mon_error("Expect index number of break point in hexadecimal.\n");
+		return false;
+	}
+
+	if (mon_token != T_END) {
+		mon_error("Too many arguments");
+		return false;
+	}
+
+	if (*index_ptr > break_point_set.size()) {
+		mon_error("Illegal Index Number!");
+		return false;
+	}
+
+	return true;
+}
+
+
+/*
+ * Remove Break Point
+ */
+void break_point_remove(void)
+{
+	uintptr index;
+
+	if (!validate_index(&index, active_break_points))
+		return;
+
+	if (0 == index) {
+		active_break_points.clear();
+		printf("Removed all break points!\n");
+		return;
+	}
+
+	BREAK_POINT_SET::iterator it = active_break_points.begin();
+	std::advance(it, index - 1);
+	// Remove break point
+	printf("Removed break point %4x at address %08lx\n", index, *it);
+	active_break_points.erase(it);
+}
+
+
+/*
+ * Disable Break Point
+ */
+void break_point_disable(void)
+{
+	uintptr index;
+
+	if (!validate_index(&index, active_break_points))
+		return;
+
+	if (0 == index) {
+		for (BREAK_POINT_SET::iterator it = active_break_points.begin(); it != active_break_points.end(); it++)
+			disabled_break_points.insert(*it);
+		active_break_points.clear();
+		printf("Disabled all break points!\n");
+		return;
+	}
+
+	BREAK_POINT_SET::iterator it = active_break_points.begin();
+	std::advance(it, index - 1);
+	// Add to disable break points
+	printf("Disabled break point %4x at address %08lx\n", index, *it);
+	disabled_break_points.insert(*it);
+	// Remove break point
+	active_break_points.erase(it);
+}
+
+
+/*
+ * Enable Break Point
+ */
+void break_point_enable(void)
+{
+	uintptr index;
+
+	if (!validate_index(&index, disabled_break_points))
+		return;
+
+	if (0 == index) {
+		for (BREAK_POINT_SET::iterator it = disabled_break_points.begin(); it != disabled_break_points.end(); it++)
+			active_break_points.insert(*it);
```

Comment:
I think you can replace this loop by:

active_break_points.insert(active_break_points.end(), disabled_break_points.begin(), disabled_break_points.end());

Reply:
I got what you mean.  Done!